### PR TITLE
Re-enable hub test

### LINKED ISSUE #1790: Enable gekhubfeaturemembership fixture test
### Checklist

- [X] I did not find a related open enhancement request.
- [X] I understand that enhancement requests filed in the GitHub repository are by default low priority.
- [X] If this request is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

### Describe the feature or resource

The current test for `TestCreateNoChangeUpdateDelete/gkehub/basic-gkehubfeaturemembership` is not working. 
We need to enable the test for the newly added Scifi controller so we have more confidence.

https://screenshot.googleplex.com/7sqp73hCpwwpQDw

### Additional information

_No response_

### Importance

_No response_

### PR DESCRIPTION
**Pull request by ziyue-101**
### Change description

Fixes  #1790

### Tests you have done

```
{"severity":"info","timestamp":"2024-05-17T23:48:38.845Z","msg":"done processing request","operation":"UPDATE","handler":"unknown fields validation","kind":"ContainerCluster","resource":{"name":"containercluster-auof6mzr7x3jkt2fwsuq","namespace":"auof6mzr7x3jkt2fwsuq"},"result-code":200,"result-reason":""}
{"severity":"info","timestamp":"2024-05-17T23:48:38.851Z","logger":"containercluster-controller","msg":"successfully finished reconcile","resource":{"name":"containercluster-auof6mzr7x3jkt2fwsuq","namespace":"auof6mzr7x3jkt2fwsuq"},"time to next reconciliation":"8m45.23393451s"}
    testreconciler.go:104: reconcile for ContainerCluster:auof6mzr7x3jkt2fwsuq/containercluster-auof6mzr7x3jkt2fwsuq took 4m23.703200004s, result was ({false 8m45.23393451s}, <nil>)
--- PASS: TestCreateNoChangeUpdateDelete (0.20s)
    --- PASS: TestCreateNoChangeUpdateDelete/gkehub (0.00s)
        --- PASS: TestCreateNoChangeUpdateDelete/gkehub/basic-gkehubfeaturemembership (762.35s)
PASS
{"severity":"info","timestamp":"2024-05-17T23:48:38.851Z","msg":"Stopping and waiting for non leader election runnables"}
{"severity":"info","timestamp":"2024-05-17T23:48:38.851Z","msg":"Stopping and waiting for leader election runnables"}
{"severity":"info","timestamp":"2024-05-17T23:48:38.851Z","msg":"Stopping and waiting for caches"}
{"severity":"info","timestamp":"2024-05-17T23:48:38.851Z","msg":"Stopping and waiting for webhooks"}
{"severity":"info","timestamp":"2024-05-17T23:48:38.851Z","logger":"controller-runtime.webhook","msg":"Shutting down webhook server with timeout of 1 minute"}
{"severity":"info","timestamp":"2024-05-17T23:48:38.852Z","msg":"Wait completed, proceeding to shutdown the manager"}
```

### CODE REVIEW

File: pkg/test/resourcefixture/testdata/basic/gkehub/v1beta1/gkehubfeaturemembership/update.yaml [hunk 1]
```diff
@@ -33,14 +33,7 @@ spec:
         policyDir: "config-connector"
         syncWaitSecs: "20"
         syncRev: "HEAD"
-    policyController:
```
Comment (maqiuyujoyce) [inline pkg/test/resourcefixture/testdata/basic/gkehub/v1beta1/gkehubfeaturemembership/update.yaml:36]: Looks like it is not possible for users to delete the feature membership when policyController is configured? If so, should we add additional notes/warnings in the field description in the CRD?

Meanwhile, have you got the chance to verify that the policyController field is not used in our samples or other test yamls?

**Review by maqiuyujoyce — COMMENTED** [review]
Thank you for root-causing and fixing the issue, @ziyue-101 !

/lgtm
/approve

/hold
Hold because I have a couple of follow up questions. But those are not blockers so feel free to remove the hold when you want to merge it.

File: pkg/test/resourcefixture/testdata/basic/gkehub/v1beta1/gkehubfeaturemembership/update.yaml (hunk 1 above)
Comment (ziyue-101) [reply pkg/test/resourcefixture/testdata/basic/gkehub/v1beta1/gkehubfeaturemembership/update.yaml:36]: To Clarify, Users can remove the policyController section under ACM in their CRs, but the underlying GCP PoCo resource are leaked. 

Opened two issue to track the doc and other tests cleanup work
#1846
#1848

/unhold

**Pull request merged**